added backend function gs.quantile

wrongu opened the pull request:
<!--
Thank you for opening this pull request!
-->

## Checklist

- [x] My pull request has a clear and explanatory title.
- [x] If neccessary, my code is [vectorized](https://www.geeksforgeeks.org/vectorization-in-python/).
- [x] I have added apropriate unit tests.
- [x] I have made sure the code passes all unit tests. (refer to comment below)
- [x] My PR follows [PEP8](https://peps.python.org/pep-0008/) guidelines. (refer to comment below)
- [x] My PR follows [geomstats coding style](https://github.com/geomstats/geomstats/blob/master/docs/contributing.rst#coding-style-guidelines) and API.
- [x] My code is properly documented and I made sure the documentation renders properly. ([Link](https://github.com/geomstats/geomstats/blob/master/docs/contributing.rst#documentation))
- [x] I have linked to issues and PRs that are relevant to this PR.


<!-- For checking consistency of entire codebase
First, run the tests related to your changes. For example, if you changed something in geomstats/spd_matrices_space.py:
$ pytest tests/tests_geomstats/test_spd_matrices.py

and then run the tests of the whole codebase to check that your feature is not breaking any of them:
$ pytest tests/

This way, further modifications on the code base are guaranteed to be consistent with the desired behavior. Merging your PR should not break any test in any backend (numpy, tensorflow or pytorch)."

For testing in alternative backends such as `numpy`, `pytorch`, `autograd`, `tensorflow` set the environment variable using:
$ export GEOMSTATS_BACKEND=<backend_name>

Next, import the `backend` module using:
import geomstats.backend as gs
-->


<!-- For flake8 tests
Install dependencies
$ pip3 install -r dev-requirements.txt

Then run the following commands:
$ flake8 --ignore=D,W503,W504 geomstats examples tests   #shadows .flake8
$ flake8 geomstats/geometry geomstats/learning           #passed two subfolders
-->

## Description

Simply adding `gs.quantile`, mirroring `np.quantile`, `torch.quantile`, and `tfp.stats.percentile`.

## Additional context

I am using this in a project of mine because I am constructing SPD matrices using a kernel, and automatically setting the length scale of the kernel according to the median distance between points.

ninamiolane commented:
Thanks for your contribution! Two things before we can merge:
- Some failed tests: FAILED tests/tests_geomstats/test_backends.py::TestBackends::test_quantile - ...
- The Linting failed (order of imports).

wrongu commented:
Strange that autograd failed before – with no changes, it passes for me locally. Tensorflow was implemented incorrectly though (now fixed).

ninamiolane commented:
Thanks @wrongu ! Btw, since this is your first PR I need to approve the GitHub actions runs, which takes time. However, for next ones, this will be done automatically -- thus less tedious. Thanks for your patience!

wrongu commented:
No rush. And sorry for the large number of micro-commits while I learned your workflow – I just fixed the (final?) linting error and squashed all back down to a single commit.

ninamiolane commented:
Failing tests are not related to this PR, merging! Thanks @wrongu !!
